source/button.js

```diff
@@ -28,200 +30,195 @@ LikelyButton.prototype = {
     init: function () {
         this.detectService();
         this.detectParams();
-        
+
         if (this.service) {
             this.initHtml();
-    
+
             setTimeout(this.initCounter.bind(this), 0);
         }
     },
-    
+
     /**
      * Update the counter
-     * 
+     *
      * @param {Object} options
      */
     update: function (options) {
         var className = '.' + config.prefix + 'counter';
-            counters  = dom.findAll(className, this.widget);
-        
-        utils.extend(this.options, utils.merge({forceUpdate: false}, options));
+        var counters = dom.findAll(className, this.widget);
+
+        utils.extend(this.options, utils.merge({ forceUpdate: false }, options));
         utils.toArray(counters).forEach(function (node) {
             node.parentNode.removeChild(node);
         });
-        
+
         this.initCounter();
     },
-    
+
     /**
      * Get the config.name of service and its options
      */
     detectService: function () {
-        var widget  = this.widget,
-            service = utils.getDataset(widget).service;
-        
+        var widget = this.widget;
+        var service = utils.getDataset(widget).service;
+
         if (!service) {
-            var classes = widget.className.split(' ');
-            
-            for (var i = 0; i < classes.length; i++) {
-                if (classes[i] in services) break;
-            }
-            
-            service = classes[i];
+            service = Object.keys(services).filter(function (service) {
+                return widget.classList.contains(service);
+            })[0];
```

Comment:
It [isn’t supported in IE](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find), and we aim to support IE10+ (#26).